info-creator/io500_add_osinfo.py

```diff
@@ -63,10 +83,15 @@ def info(key, val, unit = ""):
   info("Supercomputer.Nodes.distribution version", kv["VERSION_ID"])
 
 # Try to find country code using a reverse address
-res = subprocess.check_output("wget https://pbxbook.com/other/where_ip.html -O ip.html", shell=True, universal_newlines=True).strip()
-with open("ip.html") as f:
-  m = re.search("public IP:.*Country:</b> .* / (.*) /", f.read())
-  if m:
-    info("nationality", m.group(1))
+
+if not os.path.exists("ip.html"):
+  try:
+    res = subprocess.check_output("wget https://pbxbook.com/other/where_ip.html -O ip.html", shell=True, universal_newlines=True).strip()
```

Comment:
I think some sites might be concerned with the script going out and geo-locating their supercompter in the background.  I'd either add a prompt to ask if this is OK before running it, or just ask for user input of the two-letter ISO country code ("UK", "US", "CA", etc).  Alternately, for many systems this could be extracted from the host's domain name (e.g. check if the hostname ends in a two-character country code, capitalize it, and offer that as the default, and assume ".com" and ".edu" is the same as ".us") rather than accessing an external server.

Reply:
I think the descriptions that follow up how to use such tools can mention how it works. Nobody is forced to use it. Agreed, that alternatives are possible and they could be integrated. Btw. the schema assumes 3-letter country code not two letters making it not directly possible for translation of hostnames.